File: smart-society-frontend/src/services/visitorService.js
import API from "../api/axiosConfig";

export const getAllVisitors = () => {
  return API.get("/visitors");
};

export const addVisitor = (data) => {
  return API.post("/visitors", data);
};

export const generateOtp = (id) => {
  return API.post(`/visitors/${id}/otp`);
};

export const verifyOtp = (id, otp) => {
  return API.post(`/visitors/${id}/verify?otp=${otp}`);
};

File: SmartSociety/src/main/java/com/example/SmartSociety/service/VisitorService.java
package com.example.SmartSociety.service;



import java.util.List;

import com.example.SmartSociety.entity.Visitor;

public interface VisitorService {

    Visitor addVisitor(Visitor visitor);

    List<Visitor> getAllVisitors();

    List<Visitor> getVisitorsByResident(Long residentId);

    Visitor generateOtp(Long visitorId);

    boolean verifyOtp(Long visitorId, String otp);
    void deleteVisitor(Long id);
}
